src/zkevm_specs/evm/execution/addmod.py

```diff
@@ -0,0 +1,65 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from zkevm_specs.util import FQ, RLC
+
+# Returns 1 when a is lower than b, 0 otherwise
+def lt_u256(instruction: Instruction, a: RLC, b: RLC) -> FQ:
+    # decode RLC to bytes for a and b
+    a8s = a.le_bytes
+    b8s = b.le_bytes
+
+    a_lo = instruction.bytes_to_fq(a8s[:16])
+    a_hi = instruction.bytes_to_fq(a8s[16:])
+    b_lo = instruction.bytes_to_fq(b8s[:16])
+    b_hi = instruction.bytes_to_fq(b8s[16:])
```

Comment:
We should constraint also that `a_lo` added to `a_hi << 16` == `a` right? Same for `b`. As otherwise there's nothing constraining them.

Reply:
@CPerezz, there's any efficient way to do the `a_lo + a_hi << 16 == a`? Is this not very expensive, it will generate a lot of constraints if we use RLC here.

What about just checking the `bytes_to_fq` result ? so that $$a_{lo} = \sum_{0 \leq i < 128}  a8s[i] * 2^i$$?

Reply:
@CPerezz, there's any efficient way to do the `a_lo + a_hi << 16 == a`? Is this not very expensive, it will generate a lot of constraints if we use RLC here.

What about just checking the `bytes_to_fq` result ? so that $$a_{lo} = \sum_{0 \leq i < 128}  a8s[i] * 2^i$$?

Reply:
Well there's a cheap way to do `a_lo + a_hi << 16 == a` but requires a fixed column.

If you do:
```
Fixed Column -> 1<<16
Advice Column -> a_lo, a_hi

constraint: a_lo + (a_hi * 1<<16) - a == 0
```

Reply:
The expr I purpose has degree = 1. Which should be much cheaper than doing sum of multiplications.

Anyway, I might be missunderstanding something. So please, let me know!

Reply:
but what about the overflow? if `a_hi = 0xfff...ffff` then `a_hi * 1<<16` will overflow FQ, isn't it?

Reply:
IIUC @ed255 you mean that we can be sure no overflow happens? 
Or are you talking about doing something different than what I purposed above?

Reply:
> IIUC @ed255 you mean that we can be sure no overflow happens? Or are you talking about doing something different than what I purposed above?

What i propose is that there is no need to add a constraint for `a_lo + a_hi << 8*16 == a` (and the same for `b`), because the constraints to get `a_lo` and `a_hi` from the bytes are enough to guarantee that `a_lo` and `a_hi` are correct.